openssl/src/ssl/mod.rs

```diff
@@ -3114,6 +3112,178 @@ impl SslRef {
         }
         Ok(())
     }
+
+    #[cfg(not(boringssl))]
+    /// Sets a new default TLS/SSL method for SSL objects
+    pub fn set_method(&mut self, method: SslMethod) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::SSL_set_ssl_method(self.as_ptr(), method.as_ptr()))?;
+        };
+        Ok(())
+    }
+
+    #[corresponds(SSL_use_Private_Key_file)]
+    /// Loads the private key from a file.
+    pub fn set_private_key_file<P: AsRef<Path>>(
+        &mut self,
+        path: P,
+        ssl_file_type: SslFiletype,
+    ) -> Result<(), ErrorStack> {
+        let p = path.as_ref().as_os_str().to_str().unwrap();
+        let key_file = CString::new(p).unwrap();
+        unsafe {
+            cvt(ffi::SSL_use_PrivateKey_file(
+                self.as_ptr(),
+                key_file.as_ptr(),
+                ssl_file_type.as_raw(),
+            ))?;
+        };
+        Ok(())
+    }
+
+    #[corresponds(SSL_use_PrivateKey)]
+    /// Sets the private key.
+    pub fn set_private_key(&mut self, pkey: PKey<Private>) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::SSL_use_PrivateKey(self.as_ptr(), pkey.as_ptr()))?;
+        };
+        Ok(())
+    }
+
+    #[corresponds(SSL_use_certificate)]
+    #[cfg(any(ossl110, libressl332))]
```

Comment:
Given the above change I think you can delete this too as `SSL_use_certificate` exists in all versions of OpenSSL.